src/components/Pages/RequestPage/RequestPage.tsx

```diff
@@ -2,62 +2,160 @@ import { useCallback, useEffect, useState } from 'react'
+            <Button
+              className={styles.yesButton}
+              onClick={async () => {
+                const provider = await connection.getProvider()
+                const browserProvider = new ethers.BrowserProvider(provider)
+                const signer = await browserProvider.getSigner()
+                const signature = await signer.signMessage(request.params[0])
+                const result = await authServerFetch('outcome', { requestId, sender: await signer.getAddress(), result: signature })
+
+                if (result.error) {
+                  setResultError(result.error)
+                }
+
+                setResultSent(true)
+              }}
+            >
+              <div className={styles.yesLogo}></div> Yes, they are the same
+            </Button>
+          </div>
         </div>
-        <button
-          style={{ marginTop: '1rem' }}
-          onClick={async () => {
-            const provider = await connection.getProvider()
-            const browserProvider = new ethers.BrowserProvider(provider)
-            const signer = await browserProvider.getSigner()
-            const signature = await signer.signMessage(request.params[0])
-            await authServerFetch('outcome', { requestId, sender: await signer.getAddress(), result: signature })
-          }}
-        >
-          Sign Ephemeral Message
-        </button>
-      </div>
+      </main>
     )
   } else {
     return (
-      <div style={{ margin: '1rem' }}>
-        <h1>{request.method}</h1>
-        <pre>{JSON.stringify(request.params)}</pre>
-        <button
-          style={{ marginTop: '1rem' }}
-          onClick={async () => {
-            const provider = await connection.getProvider()
-            const browserProvider = new ethers.BrowserProvider(provider)
-            const signer = await browserProvider.getSigner()
-            const result = await browserProvider.send(request.method, request.params)
-            await authServerFetch('outcome', { requestId, sender: await signer.getAddress(), result })
-          }}
-        >
-          Execute
-        </button>
-      </div>
+      <main className={styles.main}>
+        <div className={styles.left}>
+          <div className={styles.logo}></div>
+          <div className={styles.title}>Remote Wallet Interaction</div>
+          <div className={styles.description}>The desktop app is trying to interact with your wallet.</div>
+          <div className={styles.buttons}>
+            <Button
+              className={styles.noButton}
+              onClick={() => {
+                setDenied(true)
+              }}
+            >
+              <div className={styles.noLogo}></div>Deny
+            </Button>
+            <Button
+              className={styles.yesButton}
+              onClick={async () => {
+                const provider = await connection.getProvider()
+                const browserProvider = new ethers.BrowserProvider(provider)
```

Comment:
What if we extract this section of code into a `getProvider` method to re-use it?
```suggestion
                const provider = await connection.getProvider()
                const browserProvider = new ethers.BrowserProvider(provider)
```